src/components/leftSidebar/DynamicDropdown.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useState } from "react";
+import { AutocompleteValue } from "@mui/base";
+import { InputDropdown, DropdownMenu, SDSAutocompleteOnChange } from "@czi-sds/components";
+import config from "../../../CONFIG.ts";
+
+const showDefaultAttributes = config.settings.showDefaultAttributes;
+
+export type Option = {
+    name: string;
+    label: number;
+    type: "default" | "categorical" | "continuous";
+    action: "default" | "calculate" | "provided" | "provided-normalized";
+    numCategorical: number | undefined;
+};
+
+interface DropdownProps {
+    options: Option[];
+    onClick: (event: string) => void;
+}
+
+// Define a constant for the default list of options
+const DEFAULT_DROPDOWN_OPTIONS: Option[] = [
+    { name: "uniform", label: 0, type: "default", action: "default", numCategorical: undefined },
+    { name: "x-position", label: 1, type: "continuous", action: "calculate", numCategorical: undefined },
+    { name: "y-position", label: 2, type: "continuous", action: "calculate", numCategorical: undefined },
+    { name: "z-position", label: 3, type: "continuous", action: "calculate", numCategorical: undefined },
+    { name: "sign(x-pos)", label: 4, type: "categorical", action: "calculate", numCategorical: 2 },
+    { name: "quadrants", label: 4, type: "categorical", action: "calculate", numCategorical: 8 },
```

Comment:
I'm assuming these two options should not have the same label.

```suggestion
    { name: "quadrants", label: 5, type: "categorical", action: "calculate", numCategorical: 8 },
```